File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/ClientCredential.java
package work.socialhub.kmastodon.entity;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * @author hecateball
 */
public class ClientCredential implements Serializable {

    @SerializedName("id")
    private String id;
    @SerializedName("redirect_uri")
    private String redirectUri;
    @SerializedName("client_id")
    private String clientId;
    @SerializedName("client_secret")
    private String clientSecret;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getRedirectUri() {
        return redirectUri;
    }

    public void setRedirectUri(String redirectUri) {
        this.redirectUri = redirectUri;
    }

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }

    public void setClientSecret(String clientSecret) {
        this.clientSecret = clientSecret;
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/ClientCredential.kt
package work.socialhub.kmastodon.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class ClientCredential {

    @SerialName("id")
    var id: String? = null

    @SerialName("redirect_uri")
    var redirectUri: String? = null

    @SerialName("client_id")
    var clientId: String? = null

    @SerialName("client_secret")
    var clientSecret: String? = null
}
